test/functional/repository/upload_test.rb

```diff
@@ -59,6 +59,74 @@
     File.delete("test.rpm")
   end
 
+  it "uploads srpm with content-type" do
+    File.new("test.src.rpm", "w")
+
+    params = ["--id=#{repo_id}", '--path=./test.src.rpm', '--content-type=srpm']
+
+    ex = api_expects(:content_uploads, :create, "Create upload for content") do |par|
+      par[:repository_id] == repo_id
+    end
+
+    ex.returns(upload_response)
+
+    ex2 = api_expects(:repositories, :import_uploads, 'Take in an upload') do |par|
```

Comment:
Not required to change, but there's a nicer way to do this that gives better test failure output:

```rb
    ex2 = api_expects(:repositories, :import_uploads, 'Take in an upload').with_params(...)
```

Reply:
:+1: I will go through and convert all the current tests plus the ones I added to this new format and clean up the code a bit.

Reply:
I don't mind merging this as is and having additional PRs to refactor the tests

Reply:
Since this needs this pr from Katello https://github.com/Katello/katello/pull/8099 that gives me time to work on the tests so I am fine with that :)